src/pages/index.js

```diff
@@ -43,45 +23,184 @@ const enableValidation = (config) => {
   const formList = Array.from(document.querySelectorAll(config.formSelector));
   formList.forEach((formElement) => {
     const validator = new FormValidator(config, formElement);
-    // here you get the name of the form
+    // get the name of the form
     const formName = formElement.getAttribute("name");
 
-    // here you store a validator by the `name` of the form
+    // store a validator by the `name` of the form
     formValidators[formName] = validator;
     validator.enableValidation();
   });
 };
 
 enableValidation(config);
 
+/* ----------------------- */
+/*     Class Instances     */
+/* ----------------------- */
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "6d3d8659-087c-48a4-9b8b-b1f3f711b21d",
+    "Content-Type": "application/json",
+  },
+});
+
+const userInfo = new UserInfo(
+  "#profile__name",
+  "#profile__description",
+  "#profile__image"
+);
+
+const editPopup = new PopupWithForm("#profile-popup", (obj) => {
+  handleProfileEditSubmit(obj);
+});
+
+const addPopup = new PopupWithForm("#photo-add-popup", (obj) => {
+  handlePhotoAddSubmit(obj);
+});
+
+const changePopup = new PopupWithForm("#profile-photo-popup", (obj) => {
+  handleProfilePhotoSubmit(obj);
+});
+
+const photoPopup = new PopupWithImage("#full-photo-popup");
+
+const deletePopup = new PopupWithConfirmation("#photo-delete-popup");
+
 /* ------------------ */
 /*      Functions     */
 /* ------------------ */
+function handleProfileEditSubmit(obj) {
+  editPopup.renderLoading(true, "Save");
+  api
+    .editProfile(obj)
+    .then((obj) => {
+      userInfo.setUserInfo(obj);
+      editPopup.close();
```

Comment:
The api request looks great! Everything are on the right places

Reply:
Thank you!